examples/extensions-zalando/src/main/java/org/zalando/intellij/swagger/examples/extensions/zalando/validator/zally/ZallyYamlFileValidator.java

```diff
@@ -62,36 +64,45 @@ private boolean hasZallyUrl() {
-                                       final boolean isOnTheFly,
-                                       final LintingResponse lintingResponse,
-                                       final PsiFile file) {
+    private ProblemDescriptor[] createProblems(final InspectionManager manager,
+                                               final boolean isOnTheFly,
+                                               final LintingResponse lintingResponse,
+                                               final PsiFile file) {
         final List<ProblemDescriptor> problems = Lists.newArrayList();
 
-        final Optional<PsiElement> root = getRootElement(file);
+        final List<Violation> violations = lintingResponse.getViolations();
 
-        root.filter(el -> el instanceof YAMLKeyValue)
-                .map(YAMLKeyValue.class::cast)
-                .ifPresent((psiElement) -> {
-                    final PsiElement key = psiElement.getKey();
+        for (Violation violation : violations) {
+            final Optional<YAMLKeyValue> psiElement = getPsiElement(violation.getPointer(), file);
 
-                    if (key != null) {
-                        final List<Violation> violations = lintingResponse.getViolations();
+            psiElement.ifPresent(el -> {
+                final String descriptionTemplate = String.format("[%s] %s. %s",
+                        violation.getViolationType(),
+                        violation.getTitle(),
+                        violation.getDescription());
 
-                        for (Violation violation : violations) {
-                            final String descriptionTemplate = String.format("[%s] %s Location: (%s)",
-                                    violation.getViolationType(),
-                                    violation.getTitle(),
-                                    violation.getPaths().stream().collect(Collectors.joining(", ")));
+                final PsiElement key = el.getKey();
 
-                            problems.add(manager.createProblemDescriptor(psiElement.getKey(), descriptionTemplate,
-                                    isOnTheFly, LocalQuickFix.EMPTY_ARRAY, ProblemHighlightType.GENERIC_ERROR));
-                        }
-                    }
-                });
+                if (key != null) {
+                    problems.add(manager.createProblemDescriptor(key, descriptionTemplate,
+                            isOnTheFly, LocalQuickFix.EMPTY_ARRAY, ProblemHighlightType.GENERIC_ERROR));
+                }
+            });
+        }
 
         return problems.toArray(ProblemDescriptor.EMPTY_ARRAY);
     }
 
-
+    private Optional<YAMLKeyValue> getPsiElement(final String jsonPointer, final PsiFile file) {
+        Optional<Optional<PsiElement>> psiElement = Optional.ofNullable(jsonPointer)
+                .map(pointer -> pointer
+                        .replace("/", ".")
+                        .replace("~1", "/"))
```

Comment:
We should probably also include `~0` to make the mapping complete. ([related StackOverflow question/answer](https://stackoverflow.com/questions/31483170/purpose-of-tilde-in-json-pointer))